pkg/k8sutil/portforward.go

```diff
@@ -326,14 +327,22 @@ func PortForward(localPort int, remotePort int, namespace string, getPodName fun
 
 					serviceStopCh, err := ServiceForward(clientset, cfg, desiredAdditionalPort.LocalPort, desiredAdditionalPort.ServicePort, namespace, desiredAdditionalPort.ServiceName)
 					if err != nil {
-						log.Error(errors.Wrap(err, fmt.Sprintf("failed to execute kubectl port-forward -n %s svc/%s %d:%d", namespace, desiredAdditionalPort.ServiceName, desiredAdditionalPort.LocalPort, desiredAdditionalPort.ServicePort)))
+						prevServiceForwardErr := prevServiceForwardErrMap[desiredAdditionalPort.ServiceName]
+						if prevServiceForwardErr == nil || prevServiceForwardErr.Error() != err.Error() {
+							log.Error(errors.Wrap(err, fmt.Sprintf("failed to execute kubectl port-forward -n %s svc/%s %d:%d", namespace, desiredAdditionalPort.ServiceName, desiredAdditionalPort.LocalPort, desiredAdditionalPort.ServicePort)))
+						}
+						prevServiceForwardErrMap[desiredAdditionalPort.ServiceName] = err
 						continue // try again
 					}
 					if serviceStopCh == nil {
 						// we didn't do the port forwarding, probably because the pod isn't ready.
 						// try again next loop
 						// The API doesn't return ports that aren't ready, so this is possibly rbac?
-						log.Error(errors.Errorf("failed to forward port. check that you have permission to run kubectl port-forward -n %s svc/%s %d:%d", namespace, desiredAdditionalPort.ServiceName, desiredAdditionalPort.LocalPort, desiredAdditionalPort.ServicePort))
+						prevServiceForwardErr := prevServiceForwardErrMap[desiredAdditionalPort.ServiceName]
+						if prevServiceForwardErr == nil || prevServiceForwardErr.Error() != err.Error() {
+							log.Error(errors.Errorf("failed to forward port. check that you have permission to run kubectl port-forward -n %s svc/%s %d:%d", namespace, desiredAdditionalPort.ServiceName, desiredAdditionalPort.LocalPort, desiredAdditionalPort.ServicePort))
+						}
+						prevServiceForwardErrMap[desiredAdditionalPort.ServiceName] = err
```

Comment:
will this always be set to `nil` since `err` must be `nil` to get here?  if so, then will the above log will always be printed?

Reply:
oh, this would have been a run time err when err is nil. Fixed in the next commit.
https://github.com/replicatedhq/kots/pull/3742/commits/4c0fb3119c78a75ae037a532b90d201ac6b781ad